### PR TITLE
updaters: handle "unchanged" better

### PR DESCRIPTION
**Pull request by hdonnay**
The debian and ubunut updaters weren't actually checking the passed-in fingerprint, which was resulting in spurious updates in the database.

The alpine updater wasn't using an etag, which the remote server seems to support, so this change uses that.

### CODE REVIEW

**Comment by hdonnay** [comment]
still need to spot-check the other updaters

File: alpine/fetcher.go [hunk 1]
```diff
@@ -58,18 +58,16 @@ func (u *Updater) Fetch(ctx context.Context, hint driver.Fingerprint) (io.ReadCl
 
 	var r io.Reader = res.Body
 	if _, err := io.Copy(tf, r); err != nil {
+		tf.Close()
```
Comment (ldelossa) [inline alpine/fetcher.go:61]:
```suggestion
		if tf != nil {
		  tf.Close()
		}
```

I've gotten into the habit of checking nil of these things first, may not be necessary.

File: pkg/ovalutil/fetcher.go [hunk 2]
```diff
@@ -144,12 +145,32 @@ func (f *Fetcher) Fetch(ctx context.Context, hint driver.Fingerprint) (io.ReadCl
...
+func (f fingerprint) Set(h http.Header) {
+	if f.Etag != "" {
+		h.Set("if-none-match", f.Etag)
+	}
+	if f.Date != "" {
+		h.Set("if-modified-since", f.Date)
+	}
+}
+
+func (f *fingerprint) From(h http.Header) {
```
Comment (ldelossa) [inline pkg/ovalutil/fetcher.go:168]: ```FromHeader``` ?

File: alpine/fetcher.go (hunk 1 above)
Comment (hdonnay) [reply alpine/fetcher.go:61]: In this case, tmp.File works such that if it didn't return an error, it can't return a nil pointer.

**Pull request merged**